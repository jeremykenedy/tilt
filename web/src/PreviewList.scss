@import "constants";

.PreviewList {
  margin-top: $tabnav-height;
  margin-right: $sidebar-width;
  margin-bottom: $statusbar-height;
  padding: $spacing-unit;
  width: 100%;
}

.PreviewList--empty {
  display: flex;
  justify-content: center;
}

.PreviewList ul {
  list-style: none;
  display: grid;
  grid-column: 1 / 3;
  grid-row: 5 / 6;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: $spacing-unit;
}

.PreviewList-link {
  display: flex;
  border: 2px solid rgba($color-gray-light, $translucent-ish);
  padding-top: $spacing-unit;
  padding-bottom: $spacing-unit;
  width: 100%;
  align-items: center;
  justify-content: center;
  font-family: $font-sans-serif;
  background-color: $color-gray;
  border-radius: $spacing-unit / 4;
  text-decoration: none;
  box-shadow: 2px 2px 1px 1px rgba($color-gray-darkest, $translucent);
}

.PreviewList-link:hover {
  background-color: rgba($color-gray-light, $translucent);
}
