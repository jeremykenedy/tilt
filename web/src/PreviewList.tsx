import React, { PureComponent } from "react"
import { ReactComponent as LogoWordmarkSvg } from "./assets/svg/logo-wordmark-gray.svg"
import PathBuilder from "./PathBuilder"
import { Link } from "react-router-dom"
import "./PreviewList.scss"

type PreviewListProps = {
  pathBuilder: PathBuilder
  resourcesWithEndpoints: Array<string>
}

class PreviewList extends PureComponent<PreviewListProps> {
  render() {
    let pb = this.props.pathBuilder

    let endpointsEmptyEl = (
      <section className="Pane-empty-message">
        <LogoWordmarkSvg />
        <h2>No Endpoints Found</h2>
        <p>
          If you'd like to preview your resources in the browser, <br />
          the <a href="https://docs.tilt.dev/tutorial.html">
            Tilt Tutorial
          </a>{" "}
          has more on setting up port forwarding.
        </p>
      </section>
    )

    let endpointsEl = (
      <ul>
        {this.props.resourcesWithEndpoints.map(r => (
          <li key={r}>
            <Link to={pb.path(`/r/${r}/preview`)} className="PreviewList-link">
              {r}
            </Link>
          </li>
        ))}
      </ul>
    )

    let noEndpoints = this.props.resourcesWithEndpoints.length === 0

    return (
      <section
        className={`PreviewList ${noEndpoints ? "PreviewList--empty" : ""}`}
      >
        {noEndpoints ? endpointsEmptyEl : endpointsEl}
      </section>
    )
  }
}

export default PreviewList
