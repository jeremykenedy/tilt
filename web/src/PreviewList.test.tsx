import React from "react"
import PreviewList from "./PreviewList"
import renderer from "react-test-renderer"
import PathBuilder from "./PathBuilder"
import { MemoryRouter } from "react-router"

it("renders a message if no resources have endpoint", () => {
  let pb = new PathBuilder("", "")

  const tree = renderer
    .create(<PreviewList resourcesWithEndpoints={[]} pathBuilder={pb} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it("renders a list of resources that have endpoints", () => {
  let pb = new PathBuilder("", "")

  const tree = renderer
    .create(
      <MemoryRouter initialEntries={["/"]}>
        <PreviewList
          resourcesWithEndpoints={["snack", "vigoda"]}
          pathBuilder={pb}
        />
      </MemoryRouter>
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
