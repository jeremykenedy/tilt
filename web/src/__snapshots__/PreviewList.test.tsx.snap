// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of resources that have endpoints 1`] = `
<section
  className="PreviewList "
>
  <ul>
    <li>
      <a
        className="PreviewList-link"
        href="/r/snack/preview"
        onClick={[Function]}
      >
        snack
      </a>
    </li>
    <li>
      <a
        className="PreviewList-link"
        href="/r/vigoda/preview"
        onClick={[Function]}
      >
        vigoda
      </a>
    </li>
  </ul>
</section>
`;

exports[`renders a message if no resources have endpoint 1`] = `
<section
  className="PreviewList PreviewList--empty"
>
  <section
    className="Pane-empty-message"
  >
    <svg>
      logo-wordmark-gray.svg
    </svg>
    <h2>
      No Endpoints Found
    </h2>
    <p>
      If you'd like to preview your resources in the browser, 
      <br />
      the 
      <a
        href="https://docs.tilt.dev/tutorial.html"
      >
        Tilt Tutorial
      </a>
       
      has more on setting up port forwarding.
    </p>
  </section>
</section>
`;
